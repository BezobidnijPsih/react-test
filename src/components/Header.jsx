import React, {Component} from 'react';
import styled from 'styled-components';
import GrommetHeader from 'grommet/components/Header';

class Header extends Component {
  render() {
    const Header = styled(GrommetHeader)`
      background: #4e5d6c !important;
    `;
    return (
      <Header colorIndex="grey-3" size="small">
        !
      </Header>
    );
  }
}

export default Header;
