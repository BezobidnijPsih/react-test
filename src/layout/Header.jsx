import React from 'react';
import {
  Navbar,
  FormGroup,
  FormControl,
  Glyphicon,
  Button,
  MenuItem,
  Dropdown,
  Col,
} from 'react-bootstrap';

class Header extends React.Component {
  render() {
    return (
      <Navbar staticTop>
        <Col md={3}>
          <Navbar.Header>
            <Navbar.Brand>Brand</Navbar.Brand>
            <Navbar.Toggle />
          </Navbar.Header>
        </Col>

        <Navbar.Collapse>
          <Navbar.Form pullLeft>
            <FormGroup>
              <FormControl type="text" placeholder="Search" />
            </FormGroup>{' '}
            <Button type="submit">Submit</Button>
          </Navbar.Form>
          <Navbar.Form pullRight>
            <Dropdown id="basic-nav-dropdown">
              <Dropdown.Toggle>
                <Glyphicon glyph="user" /> Admin
              </Dropdown.Toggle>
              <Dropdown.Menu>
                <MenuItem eventKey={1}>Profile</MenuItem>
                <MenuItem eventKey={2}>Something else</MenuItem>
                <MenuItem divider />
                <MenuItem eventKey={4}>
                  <Glyphicon glyph="log-out" /> Log out
                </MenuItem>
              </Dropdown.Menu>
            </Dropdown>
          </Navbar.Form>
        </Navbar.Collapse>
      </Navbar>
    );
  }
}

export default Header;
