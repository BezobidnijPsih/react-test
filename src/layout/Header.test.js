import React from 'react';
import toJson from 'enzyme-to-json';
import Header from './Header';

describe('Header', () => {
  test('should render correctly', () => {
    const header = shallow(<Header />);
    expect(toJson(header)).toMatchSnapshot();
  });
});
