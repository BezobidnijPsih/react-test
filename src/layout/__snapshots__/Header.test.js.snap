// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render correctly 1`] = `
<Uncontrolled(Navbar)
  bsStyle="default"
  inverse={true}
  staticTop={true}
>
  <NavbarHeader>
    <NavbarBrand>
      Brand
    </NavbarBrand>
    <NavbarToggle />
  </NavbarHeader>
  <NavbarCollapse>
    <NavbarForm
      componentClass="div"
      pullLeft={true}
      pullRight={false}
    >
      <FormGroup
        bsClass="form-group"
      >
        <FormControl
          bsClass="form-control"
          componentClass="input"
          placeholder="Search"
          type="text"
        />
      </FormGroup>
       
      <Button
        active={false}
        block={false}
        bsClass="btn"
        bsStyle="default"
        disabled={false}
        type="submit"
      >
        Submit
      </Button>
    </NavbarForm>
    <NavbarForm
      componentClass="div"
      pullLeft={false}
      pullRight={true}
    >
      <Uncontrolled(Dropdown)
        id="basic-nav-dropdown"
      >
        <DropdownToggle
          bsClass="dropdown-toggle"
          bsRole="toggle"
          open={false}
          useAnchor={false}
        >
          <Glyphicon
            bsClass="glyphicon"
            glyph="user"
          />
           User
        </DropdownToggle>
        <DropdownMenu
          bsClass="dropdown-menu"
          bsRole="menu"
          pullRight={false}
        >
          <MenuItem
            bsClass="dropdown"
            disabled={false}
            divider={false}
            eventKey={1}
            header={false}
          >
            Profile
          </MenuItem>
          <MenuItem
            bsClass="dropdown"
            disabled={false}
            divider={false}
            eventKey={2}
            header={false}
          >
            Something else
          </MenuItem>
          <MenuItem
            bsClass="dropdown"
            disabled={false}
            divider={true}
            header={false}
          />
          <MenuItem
            bsClass="dropdown"
            disabled={false}
            divider={false}
            eventKey={4}
            header={false}
          >
            <Glyphicon
              bsClass="glyphicon"
              glyph="log-out"
            />
             Log out
          </MenuItem>
        </DropdownMenu>
      </Uncontrolled(Dropdown)>
    </NavbarForm>
  </NavbarCollapse>
</Uncontrolled(Navbar)>
`;
